packages/loader/container-loader/src/container.ts

```diff
@@ -596,6 +596,32 @@ export class Container
 		return this.loader.services.codeLoader;
 	}
 
+	/**
+	 * {@inheritDoc @fluidframework/container-definitions#IContainer.entryPoint}
+	 */
+	public async getEntryPoint?(): Promise<FluidObject | undefined> {
+		if (this._lifecycleState === "closed" || this._lifecycleState === "disposed") {
+			throw new UsageError("The container is already closed or disposed");
+		}
+		while (this._context === undefined) {
```

Comment:
will this loop exist if the container is closed while we are in the loop waiting?

Reply:
The Promise will get rejected and since it's being awaited that should end up throwing out of `getEntryPoint()`. 

That said, I'm changing the handlers to be for "contextChanged" and "disposed", but I have the same question as in the comment above... what exactly does it mean if the container closes before emitting "contextChanged" while we wait here, and how should we react to it. If we're saying that the entryPoint should be available for a closed Container (at least if it didn't close due to an error?), we need to guarantee that a non-error close will still set `this._context`?

@justus-camp I'm also confused by this documentation:

> States are allowed to progress to further states: "loading" - "loaded" - "closing" - "disposing" - "closed" - "disposed"
>
> For example, moving from "closed" to "disposing" is not allowed since it is an earlier state.
>
> loading: Container has been created, but is not yet in normal/loaded state 
> loaded: Container is in normal/loaded state
> closing: Container has started closing process (for re-entrancy prevention)
> disposing: Container has started disposing process (for re-entrancy prevention)
> closed: Container has closed
> disposed: Container has been disposed

I'm assuming this doesn't mean the Container will necessarily go through all of them. So if someone closes (not disposes) the Container, I'd expect lifecycle to change to "closing", then to "closed" (without passing through "disposing") in the middle. But if the Container is disposed after that, it *can't* move back to "disposing" before it moves to "disposed"?

Reply:
@alexvy86 Yes, containers may not go through every state. This is just a variable that helps prevent reentrancy as well as give some logging information.
The tricky part with this "state" was if we close then dispose. Should we then consider `"disposing"` in `this.closed`? If we don't, then container could be closed, start disposing resources, and someone could use the container in an unexpected way (which would normally be prevented by `this.closed`). The simplest option that made sense was to prevent moving from `"closed"` to `"disposing"` since it is mainly used for telemetry.